server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/executeBash.ts

```diff
@@ -441,11 +442,133 @@ export class ExecuteBash {
         }
     }
 
+    /**
+     * Check if terminal integration is available in the IDE
+     */
+    private isTerminalAvailable(): boolean {
+        try {
+            // Check if LSP connection is available
+            if (!this.features.lsp) {
+                this.logging.debug('LSP connection not available, using child process for command execution')
+                return false
+            }
+
+            // Check if client capabilities include terminalAvailability
+            const initParams = this.features.lsp.getClientInitializeParams()
+            const terminalAvailability =
+                initParams?.initializationOptions?.aws?.awsClientCapabilities?.q?.terminalAvailability
+
+            if (!terminalAvailability) {
+                this.logging.debug('Terminal support is not available, using child process for command execution')
+                return false
+            }
+
+            this.logging.debug('Terminal integration is available')
+            return true
+        } catch (error) {
+            this.logging.warn(`Error checking terminal availability: ${(error as Error).message}`)
+            return false
+        }
+    }
+
     // TODO: generalize cancellation logic for tools.
     public async invoke(
         params: ExecuteBashParams,
         cancellationToken?: CancellationToken,
         updates?: WritableStream
+    ): Promise<InvokeOutput> {
+        // Check if IDE terminal integration is available
+        const terminalAvailable = this.isTerminalAvailable()
+
+        if (terminalAvailable) {
+            this.logging.info('Using IDE terminal integration for command execution')
+            return this.executeViaIDETerminal(params, cancellationToken, updates)
+        } else {
+            this.logging.info('Using child process for command execution')
+            return this.executeViaChildProcess(params, cancellationToken, updates)
+        }
+    }
+
+    /**
+     * Execute command using IDE terminal integration
+     */
+    private async executeViaIDETerminal(
+        params: ExecuteBashParams,
+        cancellationToken?: CancellationToken,
+        updates?: WritableStream
+    ): Promise<InvokeOutput> {
+        this.logging.info(`Executing command via IDE terminal: "${params.command}" in cwd: "${params.cwd}"`)
+
+        try {
+            if (!this.features.lsp) {
+                throw new Error('LSP connection not available')
+            }
+
+            // Prepare the terminal command request
+            const terminalRequest: ExecuteTerminalCommandParams = {
+                command: params.command,
+                cwd: params.cwd,
+                terminalName: 'Amazon Q Agent',
+                createNew: false, // Reuse existing terminal if available
```

Comment:
Could this cause issues with command history or state? Should we be reusing terminals?